pycontrails/models/cocip/cocip.py

```diff
@@ -1022,22 +1043,7 @@ def _simulate_contrail_evolution(self) -> None:
                 continue
 
             # Update met, rad slices as needed
-            # We need to both interpolate latest_contrail, as well as the "contrail_2"
-            # created by calc_timestep_contrail_evolution. This "contrail_2" object
-            # has constant time at "time_end", hence the buffer we apply below.
-            # After the downwash_contrails is all used up, these updates are intended
-            # to happen once each hour
-            buffers["time_buffer"] = (
-                np.timedelta64(0, "ns"),
-                time_end - latest_contrail["time"].max(),
-            )
-            if time_end > met.indexes["time"].to_numpy()[-1]:
-                logger.debug("Downselect met at time_end %s within Cocip evolution", time_end)
-                met = latest_contrail.downselect_met(self.met, **buffers, copy=False)
-                met = add_tau_cirrus(met)
-            if time_end > rad.indexes["time"].to_numpy()[-1]:
-                logger.debug("Downselect rad at time_end %s within Cocip evolution", time_end)
-                rad = latest_contrail.downselect_met(self.rad, **buffers, copy=False)
```

Comment:
This logic doesn't work when ``preprocess_mem=False``. In this case, ``self._process_downwash_flight`` doesn't return a variable ``met`` downselected to cover ``self._downwash_contrail``, so you will have to downselect within the timestepping loop before all of the points in ``self._downwash_contrail`` are consumed. In this case it's no longer sufficient just to downselect to cover ``latest_contrail``; you also have to make sure you cover all the points in ``self._downwash_contrail`` that *will* be consumed within in the hour. This makes the downselection logic significantly more complicated, so I've extracted it to its own method.